src/io/flutter/font/FontPreviewProcessor.java

```diff
@@ -77,119 +94,304 @@ public void projectClosed(@NotNull Project project) {
         ProjectManagerListener.super.projectClosed(project);
       }
     });
+    final String projectPath = project.getBasePath();
+    final WorkItem item = new WorkItem(projectPath);
+    WORK_ITEMS.put(projectPath, item);
+
     final String packagesText = FlutterSettings.getInstance().getFontPackages();
     final String[] packages = packagesText.split(PACKAGE_SEPARATORS);
-    //com.intellij.util.SlowOperations.allowSlowOperations(() -> { // 2021.2
-      for (String pack : packages) {
-        findFontClasses(project, pack.trim());
+    item.addPackages(
+      Arrays.stream(packages)
+        .map(String::trim)
+        .filter((each) -> !each.isEmpty() || FontPreviewProcessor.UNSUPPORTED_PACKAGES.get(each) != null)
+        .collect(Collectors.toList()));
+    processItems(project);
+  }
+
+  void processItems(@NotNull Project project) {
+    final Task.Backgroundable task = new Task.Backgroundable(project, FlutterBundle.message("icon.preview.analysis"), true) {
+
+      public void run(@NotNull final ProgressIndicator indicator) {
+        final AtomicBoolean isRunning = new AtomicBoolean(true);
+        final long startTime = System.currentTimeMillis();
+        while (isRunning.get()) {
+          final String path = project.getBasePath();
+          final WorkItem item = WORK_ITEMS.get(path);
+          if (!processNextItem(project, item)) {
+            if (item.filesWithNoClasses.isEmpty()) {
+              // Finished, clean up and exit.
+              synchronized (this) {
+                if (item == WORK_ITEMS.get(path)) {
+                  WORK_ITEMS.remove(path);
+                }
+                isRunning.set(false);
+                if (System.currentTimeMillis() - startTime > 1000) {
```

Comment:
Is this time arbitrary? Would it be better to always do this?

Reply:
This is a kind of heavyweight action that can cause screen flicker. I don't think we want to do it unless there is a reason. But in my testing, with only a small amount of analysis there is no need to do it. And in my case, there was only one icon that was not displayed, and even that was displayed sometimes. Thanks for the review!